### PR TITLE
fix placeholder for loaded folders through ListRecursive

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/actions/fs/index.js [hunk 1]
```diff
@@ -151,9 +152,14 @@ function* folderList(action: FsGen.FolderListLoadPayload): Saga.SagaGenerator<an
...
-    rootPath
-  )
+  const {
+    lastModifiedTimestamp,
+    lastWriter,
+    size,
+    writable,
+    favoriteChildren,
+    tlfMeta,
+  }: Types.FolderPathItem = state.fs.pathItems.get(rootPath)
```
Comment (songgao) [inline shared/actions/fs/index.js]: I think my editor linter made these changes, but it passed the precommit hooks so I didn't bother reverting it.

File: shared/actions/fs/index.js [hunk 2]
```diff
@@ -47,6 +47,7 @@ const makeEntry = (d: RPCTypes.Dirent, children?: Set<string>) => {
       return Constants.makeFolder({
         ...direntToMetadata(d),
         children: I.Set(children),
+        progress: children ? 'loaded' : undefined,
```
Comment (songgao) [inline shared/actions/fs/index.js]: This is the fix. The container uses `progress` field to figure out whether a folder is really empty or just not loaded.

**Comment by songgao** [comment]
actually, needed to add another commit ...

**Comment by songgao** [comment]
OK ready again, but I think you've seen the gist. The last forced commit just changed `.split('/') === 1` into `.indexOf('/') < 0`.

File: shared/actions/fs/index.js [hunk 3]
```diff
@@ -146,7 +146,13 @@ function* folderList(action: FsGen.FolderListLoadPayload): Saga.SagaGenerator<an
 
   const direntToPathAndPathItem = (d: RPCTypes.Dirent) => {
     const path = Types.pathConcat(rootPath, d.name)
-    return [path, makeEntry(d, childMap.get(path))]
+    const entry = makeEntry(d, childMap.get(path))
+    if (entry.type === 'folder' && Types.getPathLevel(path) > 3 && d.name.indexOf('/') < 0) {
+      // Since we are loading with a depth of 2, first level directoryies are
```
Comment (jzila) [inline shared/actions/fs/index.js]: "directories"

**Pull request merged**